quesma/model/bucket_aggregations/ip_prefix.go

```diff
@@ -0,0 +1,204 @@
+// Copyright Quesma, licensed under the Elastic License 2.0.
+// SPDX-License-Identifier: Elastic-2.0
+package bucket_aggregations
+
+import (
+	"context"
+	"fmt"
+	"quesma/logger"
+	"quesma/model"
+	"reflect"
+)
+
+// Current limitation: we expect Clickhouse field to be IPv4 (and not IPv6)
+
+// Clickhouse table to test SQLs:
+// CREATE TABLE __quesma_table_name (clientip IPv4) ENGINE=Log
+// INSERT INTO __quesma_table_name VALUES ('0.0.0.0'), ('5.5.5.5'), ('90.180.90.180'), ('128.200.0.8'),  ('192.168.1.67'), ('222.168.22.67')
+
+// TODO make part of QueryType interface and implement for all aggregations
+// TODO add bad requests to tests
+// Doing so will ensure we see 100% of what we're interested in in our logs (now we see ~95%)
+func CheckParamsIpPrefix(ctx context.Context, paramsRaw any) error {
+	requiredParams := map[string]string{
+		"field":         "string",
+		"prefix_length": "float64", // TODO should be int, will be fixed
+	}
+	optionalParams := map[string]string{
+		"is_ipv6":              "bool",
+		"append_prefix_length": "bool",
+		"keyed":                "bool",
+		"min_doc_count":        "int",
+	}
+	logIfYouSeeThemParams := []string{"min_doc_count"}
```

Comment:
In generalized version we could log unkown parameters too.

Reply:
unknown parameter in Kibana is always an error. We also error out here if we see it, here:
```
// check if only required/optional are present
for paramName := range params {
  if _, isRequired := requiredParams[paramName]; !isRequired {
    wantedType, isOptional := optionalParams[paramName]
    if !isOptional {
      return fmt.Errorf("unexpected parameter %s found in IP Range params %v", paramName, params)
    }
```
Or you meant something else @jakozaur ?